### ISSUE TITLE
HTML extensions stripped from DOT SVG links with tagfiles

### ISSUE THREAD

**Issue by asteele0** [doxygen/doxygen#9251]
**Describe the bug**
With an external project with a tagfile, the DOT graphs created are invalid. The href values are being generated without the .html extension.

Given the screenshot Graph1, the generated files are incorrect:

**cat .\html\classinternal_1_1_internal2__coll__graph.map**
```xml
<map id="internal.Internal2" name="internal.Internal2">
  <area shape="rect" id="node1" title="class" alt="" coords="18,323,135,349"/>
  <area shape="rect" id="node2" href="$class_internal_proj.html" title="Internal proj class." alt="" coords="33,247,120,273"/>
  <area shape="rect" id="node3" href="C:/doxygen-bug/ext_1/external.tag$classext__1_1_1_external_sub_c" title=" " alt="" coords="5,171,148,197"/>
  <area shape="rect" id="node4" href="C:/doxygen-bug/ext_1/external.tag$classext__1_1_1_external_proj" title=" " alt="" coords="146,95,279,121"/>
<area shape="rect" id="node5" title=" " alt="" coords="14,5,70,32"/>
</map>
```

**cat .\html\classinternal_1_1_internal2__coll__graph.svg** (snipped)
```xml
<g id="a_node3"><a target="_parent" xlink:href="../../ext_1/html/classext__1_1_1_external_sub_c" xlink:title=" ">
```

**Note** the links are correctly generated when setting `HAVE_DOT` to `NO`:
```html
<map id="internal.Internal2_map" name="internal.Internal2_map">
  <area href="class_internal_proj.html" title="Internal proj class." alt="InternalProj" shape="rect" coords="0,112,125,136">
  <area target="_parent" href="../../ext_1/html/classext__1_1_1_external_sub_c.html" alt="ext_1::ExternalSubC" shape="rect" coords="0,56,125,80">
  <area target="_parent" href="../../ext_1/html/classext__1_1_1_external_proj.html" alt="ext_1::ExternalProj" shape="rect" coords="0,0,125,24">
</map>
```

I wasn't able to find any other posted issue that seemed to fit, apologies if I missed it.

**Expected behavior**
The generated links would properly link to the external documentation pages

**Screenshots**
Graph1:
![image](https://user-images.githubusercontent.com/76005161/161642586-719d534b-4cd1-45cb-81f4-0f6ffbc18f34.png)

**To Reproduce**
[doxygen-bug.zip](https://github.com/doxygen/doxygen/files/8413514/doxygen-bug.zip)
1. Run `doxygen ext.dox` in `ext_1`
2. Run `doxygen internal.dox` in `internal`

**Version**
doxygen --version: `1.9.3 (c0b9eafbfb53286ce31e75e2b6c976ee4d345473)`
OS: Windows 10 x64
Binary from website

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9252 

(Note to see the problem `HAVE_DOT` has to be set in the settings files provided).

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by asteele0** [comment]
Thanks for the fast turnaround!

Do you have any estimation of when a new version will be released?

**Comment by albert-github** [comment]
> Do you have any estimation of when a new version will be released?

There is no scheduled date available yet.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
